org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/impl/XtextBuilder.java

```diff
@@ -57,7 +65,31 @@
 	private static final Logger log = Logger.getLogger(XtextBuilder.class);
 
 	public static final String BUILDER_ID = XtextProjectHelper.BUILDER_ID;
-
+	/**
+	 * Name of a preference defining the scheduling rule strategy for the builder.
+	 * 
+	 * @since 2.16
+	 */
+	@Beta
+	public static final String PREF_SCHEDULING_RULE = "schedulingrule"; //$NON-NLS-1$
+	/**
+	 * Options for scheduling rules.
+	 * 
+	 * @since 2.16
+	 */
+	public static enum SchedulingOption {
+		WORKSPACE,
+		// all Xtext projects in the workspace
+		ALL_XTEXT_PROJECTS,
+		// the currently building project
+		PROJECT,
+		// null scheduling rule
+		NULL;
+	}
+	
+	private IPreferenceStore preferenceStore = new ScopedPreferenceStore(InstanceScope.INSTANCE,
```

Comment:
same here. totally unsure how to access correctly

Reply:
but this should access the InstanceScope here, shouldn't it?

Reply:
i have no idea how to correctly obtain a thing to read here. find only outdated or you can do 1000 ways documentation

Reply:
ScopedPreferenceStore is used quite common. E.g. 
org.eclipse.ui.internal.ide.ChooseWorkspaceData.readPersistedData()
org.eclipse.jdt.internal.ui.preferences.JavaBasePreferencePage.JavaBasePreferencePage()

Reply:
yes org.eclipse.ui.plugin.AbstractUIPlugin.getPreferenceStore()